### PR TITLE
ReaderGoto: go to %

### PR DESCRIPTION
**Pull request by hius07**
---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/8395)

### CODE REVIEW

**Comment by hius07** [comment]
Percent is the only way to share a location in a book with a friend who reads the same book in his own device.

File: frontend/apps/reader/modules/readergoto.lua [hunk 1]
```diff
@@ -43,8 +43,9 @@ function ReaderGoto:onShowGotoDialog()
     else
         input_hint = T("@%1 (1 - %2)", curr_page, self.document:getPageCount())
     end
+    input_hint = input_hint .. string.format("  %.2f%%", curr_page / self.document:getPageCount() * 100)
     self.goto_dialog = InputDialog:new{
-        title = _("Enter page number"),
+        title = _("Enter page number or %"),
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readergoto.lua:48]: Is there a specific rationale behind not writing percentage?

**Comment by poire-z** [comment]
> Percent is the only way to share a location in a book with a friend who reads the same book in his own device.

I would indeed have asked why/when we would need this ? :) Feels a bit rare use case though.
I rarely use this popup, and having 4 buttons makes it nicer than 3 I guess, so I guess it's fine.

But one should then enter `86%`, and not `0.86%`.
And/or can a manually added `%` in the inputted text be enough for a single button to detect it's a % and not a page number? What happens if I enter `86%` and hit `Go To page`, or if I enter `132` and hit `Go to %` ?

**Comment by hius07** [comment]
> But one should then enter `86%`, and not `0.86%`.

The hint shows current percentage (page 7 of 810).

Having 4 buttons is nice and allows to avoid entering the % sign.
(BTW one of the reasons was that we didn't have the % key in the numeric layer).

`Go to page` doesn't accept 86%.
`Go to %` goes to the first / last page if the percentage entered is out of 0-100 range.

File: frontend/apps/reader/modules/readergoto.lua (hunk 1 above)
Comment (hius07) [reply frontend/apps/reader/modules/readergoto.lua:48]: Can we keep `Go to %` in the button (lack of space)?

**Comment by poire-z** [comment]
> The hint shows current percentage (page 7 of 810).

OK, but you should still _display_ this current percentage as 86% (and not 0.86%) - if what the user wanting 45% is to input 45% (and not 0.45%) to get there.

File: frontend/apps/reader/modules/readergoto.lua (hunk 1 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readergoto.lua:48]: Of course. But if the rationale is just that it's shorter, no need in the title. ^_^

**Comment by hius07** [comment]
> not 0.86%

Some misunderstanding.
I display `0.86%` for the current page 7.
If I want to go to the page 7 of 810 I must enter 0.86 and press `Go to %`.

**Review by poire-z — APPROVED** [review]
Just be sure it works fine in scroll/continuous mode (epub and pdf).

**Comment by hius07** [comment]
Wait a little bit with this please, I have some work in it.

**Comment by hius07** [comment]
Ready

**Pull request merged**